CHANGELOG.md

```diff
@@ -4,6 +4,52 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [3.15.0] - 2024-10-29
+
+### Added
+
+* PHP 8.4 support by @heiglandreas in [#1473](https://github.com/Behat/Behat/pull/1473), @jrfnl in [#1478](https://github.com/Behat/Behat/pull/1478),
+  @jrfnl in [#1477](https://github.com/Behat/Behat/pull/1477)
+* Support config files named `behat.dist.y[a]ml` by @uuf6429 in [#1464](https://github.com/Behat/Behat/pull/1464)
+* Add a `--rerun-only` flag to immediately exit 0 without running anything if there were no failures on the previous run
+  by @Treast in [#1466](https://github.com/Behat/Behat/pull/1466)
+* Allow profiles to override extensions in the config file by @Zayon in [#1341](https://github.com/Behat/Behat/pull/1341)
+* Support configuring a preferred profile to use instead of `default` if nothing was specified at runtime
+  by @andrewnicols in [#1334](https://github.com/Behat/Behat/pull/1334)
+* Add void return type when generating new snippets by @carlos-granados in [#1463](https://github.com/Behat/Behat/pull/1463)
+
+### Fixed
+
+* Fix enforcing that 32 character (or longer) turnip pattern names are not allowed
+  by @ivastly in [#1457](https://github.com/Behat/Behat/pull/1457) and @acoulton in [#1483](https://github.com/Behat/Behat/pull/1483)
+* Fix generating the PHAR for releases by upgrading the build tooling by @heiglandreas in [#1462](https://github.com/Behat/Behat/pull/1462)
+
+### Internal
```

Comment:
Does this even belong to a changelog if it is internal ?

Reply:
It's a valid question - personally, I would probably lean closer to "no" on some of these, but thought that might be controversial with others (conscious the convention these days is increasingly to list absolutely everything). 

That said, there's always the chance someone is using the package in an unexpected way (including maybe applying a manual patch to a file as part of composer install, or something) so there might be an argument to note these since at least one changes runtime code.

It also feels a safer convention to say everything is listed, but some things might be "Internal". If we make the wrong call on a PR that just refactors things or updates code style, at least it's there somewhere rather than missing outright. Plus it shows it's been considered, rather than just overlooked.

Finally there's the increasing convention to credit contributors in the changelog - I don't want to downplay people's work just because the immediate benefit is to us rather than end-users.

I did, however, decide not to include dependabot PRs at all.

Reply:
Listing everything is often caused by the fact that it is generated automatically, without a human review step.

Reply:
I'm very aware of that - that's my motivation to find a release process that does have a review step.

Personally, I think I've stuck a balance between including everything and excess verbosity that I'd be happy with as an end user.